src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterReceiver.cs

```diff
@@ -96,7 +135,39 @@ public Task MessagesDeliveredAsync(IList<IBatchContainer> messages)
 
         private IList<IBatchContainer> BuildBatchList(IEnumerable<EventData> messages)
         {
-            return messages.Select(EventHubBatchContainer.FromEventData).ToList();
+            var batches = new List<IBatchContainer>();
+            foreach (EventData message in messages)
+            {
+                cache.Add(message);
+                batches.Add(new StreamActivityNotificationBatch(Guid.Parse(message.PartitionKey),
+                    message.GetStreamNamespaceProperty(), new EventSequenceToken(message.SequenceNumber, 0)));
+                
+            }
+            return batches;
+        }
+
+        private void Purge(IDisposable purgedResource)
+        {
+            cache.Purge(purgedResource);
+        }
+
+        private class StreamActivityNotificationBatch : IBatchContainer
```

Comment:
Is `StreamActivityNotificationBatch` supposed to be Serializable? If yes, lets mark it and also add JSON properties attributes where necessary. If not, just add a comment that it is not, since other impls. of `IBatchContainer` usually are.